src/Braspag.FluentQueryBuilder/SelectBuilder.cs

```diff
@@ -156,6 +195,28 @@ public SelectBuilder Paginated(int pageSize, int currentPage)
             return this;
         }
 
+        /// <summary>
+        /// Adds pagination using DENSE_RANK
+        /// </summary>
+        /// <param name="pageSize">Number of rows per page</param>
+        /// <param name="currentPage">Current page</param>
+        /// <param name="rankField">Field used for ranking</param>
+        public WhereBuilder PaginatedByDenseRank(int pageSize, int currentPage, string rankField)
```

Comment:
I thik that will be better if you change the method "Paginated" and add an option to select the paginated type (OffSet or DenseRank). This method may continue to exist as a private method for use in Paginated method

Reply:
Those two methods return different types.
"Paginated" returns SelectBuilder and "PaginatedByDenseRank" returns a WhereBuilder.
I can create a "ToSelectBuilder" method on WhereBuilder and make the suggested changes or leave it as it is.

Reply:
Done